### ISSUE TITLE
Failure to detect the `import typing as tt` convention

### ISSUE THREAD

**Issue by ddelange** [jsh9/pydoclint#108]
```py
import typing as tt

def read_jsonl(
    path_jsonl: str,
) -> tt.Iterator[dict[int, tuple[int, int, int]]]:
    """Read a jsonl file, one record at a time.

    Args:
        path_jsonl: Path to a jsonl file.

    Yields:
        One record
    """
    with Path(path_jsonl).open("rb") as fp:
        yield from fp
```

Will flag false positives:
```
    29: DOC201: Function `mock_loop_over_video` does not have a return section in docstring
    29: DOC403: Function `mock_loop_over_video` has a "Yields" section in the docstring, but there are no "yield" statements, or the return annotation is not a Generator/Iterator/Iterable.
```

My current hotfix is using `from typing import Iterator` directly.
